### ISSUE TITLE
body-parser errors are not handled as HttpErrors

### ISSUE THREAD

**Issue by McMerph** [RobinTail/express-zod-api#907]
### Steps to reproduce:
- use `createServer` with default `server.jsonParser` to start the server
- send request with invalid JSON. e.g.:
```
curl --location 'http://localhost:{{port}}/whatever-path' \
--header 'Content-Type: application/json' \
--data '{
    a: "b"
}'
```

### Actual behaviour:
The server responds with a 500 error

### Expected behaviour
The server responds with a 400 error

---
One possible workaround is to pass a custom `jsonParser`:
```
import { isHttpError } from "http-errors";

...
const rawJsonParser = express.json();
const jsonParser: typeof rawJsonParser = (request, response, next) => {
  rawJsonParser(request, response, (error) => {
    if (error instanceof Error) {
      response.statusCode = isHttpError(error)
        ? error.status
        : getStatusCodeFromError(error);
      response.end(getMessageFromError(error));
    } else {
      next(error);
    }
  });
};
...
createServer(createConfig({
...
server: {
  jsonParser,
  ...
}
}), routing)
```

Note that code above will not work without the `isHttpError` check, because there is no such check inside `getStatusCodeFromError`. @RobinTail, Is it intentional?

**Comment by RobinTail** [comment]
Hello @McMerph 

>  @RobinTail, Is it intentional?

Let me check... I'm sure that I actually have a test for this case...

**Comment by RobinTail** [comment]
I found that test. Now I need to recall why it's `500`.
Please stand by :)

https://github.com/RobinTail/express-zod-api/blob/d7bf42feecfd50f58af798bbfd66a6e014c89cfe/tests/system/system.spec.ts#L350

**Comment by RobinTail** [comment]
The test is 2 years old and is from the version 0.1.2.
I believe it's a bug.

**Comment by RobinTail** [comment]
@McMerph ,
this is a very long-living issue.
Thank you very much for finding it.

🏅

**Comment by RobinTail** [comment]
🚀 The fix released in v10.1.2
